sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/CosmosPagedFluxTest.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ *
+ */
+
+package com.azure.cosmos;
+
+import com.azure.cosmos.implementation.InternalObjectNode;
+import com.azure.cosmos.models.CosmosQueryRequestOptions;
+import com.azure.cosmos.rx.TestSuiteBase;
+import com.azure.cosmos.util.CosmosPagedFlux;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Factory;
+import org.testng.annotations.Test;
+
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class CosmosPagedFluxTest extends TestSuiteBase {
+
+    private static final int NUM_OF_ITEMS = 10;
+
+    private CosmosAsyncClient cosmosAsyncClient;
+    private CosmosAsyncContainer cosmosAsyncContainer;
+
+    @Factory(dataProvider = "clientBuilders")
+    public CosmosPagedFluxTest(CosmosClientBuilder clientBuilder) {
+        super(clientBuilder);
+    }
+
+    @BeforeClass(groups = { "simple" }, timeOut = SETUP_TIMEOUT)
+    public void before_CosmosPagedFluxTest() {
+        assertThat(this.cosmosAsyncClient).isNull();
+        this.cosmosAsyncClient = getClientBuilder().buildAsyncClient();
+        CosmosAsyncContainer asyncContainer = getSharedMultiPartitionCosmosContainer(this.cosmosAsyncClient);
+        cosmosAsyncContainer =
+            cosmosAsyncClient.getDatabase(asyncContainer.getDatabase().getId()).getContainer(asyncContainer.getId());
+        createItems(NUM_OF_ITEMS);
+    }
+
+    @AfterClass(groups = { "simple" }, timeOut = SHUTDOWN_TIMEOUT, alwaysRun = true)
+    public void afterClass() {
+        assertThat(this.cosmosAsyncClient).isNotNull();
+        this.cosmosAsyncClient.close();
+    }
+
+    @Test(groups = { "simple" }, timeOut = TIMEOUT)
+    public void readAllItemsByPageWithCosmosPagedFluxHandler() throws Exception {
+        CosmosQueryRequestOptions cosmosQueryRequestOptions = new CosmosQueryRequestOptions();
+
+        CosmosPagedFlux<InternalObjectNode> cosmosPagedFlux =
```

Comment:
for the new test please don't use `InternalObjectNode` this will be used by the end user.

Please rely on ObjectNode or a simple POJO.